src/devices/machine/am9513.cpp

```diff
@@ -1423,19 +1434,20 @@ u16 am9513_device::read16(offs_t offset)
 void am9513_device::write16(offs_t offset, u16 data)
 {
 	if ((!bus_is_16_bit() || BIT(offset, 0)) && (data & 0xff00) != 0xff00)
-		logerror("Errant write of %02X to upper byte of %s register in %d-bit bus mode\n",
+		LOGWARN("Errant write of %02X to upper byte of %s register in %d-bit bus mode\n",
 				(data & 0xff00) >> 8,
 				BIT(offset, 0) ? "control" : "data",
 				bus_is_16_bit() ? 16 : 8);
 
 	if (BIT(offset, 0))
 	{
+		command_write(data & 0x00ff);
+
+		// NB testing afterwards because this command may have been changing bus width
 		if ((data & 0x00ff) == 0x00e7)
-			logerror("8-bit data bus selected with 16-bit write\n");
+			LOGWARN("8-bit data bus selected with 16-bit write\n");
 		else if ((data & 0x00ff) == 0x00ef && !bus_is_16_bit())
-			logerror("16-bit data bus selected\n");
-
-		command_write(data & 0x00ff);
+			LOGWARN("16-bit data bus selected\n");
```

Comment:
I think this won’t work properly after the change – `else if ((data & 0x00ff) == 0x00ef && !bus_is_16_bit())` won’t be true after calling `command_write` because the 16-bit mode will have selected.